calico/felix/frules.py

```diff
@@ -538,7 +538,7 @@ def update_ipsets(type,
     """
     for rule in rule_list:
         if rule.get('cidr') is None:
-            log.error("Invalid %s rule without cidr for %s : %s",
+            log.error("Invalid %s rule without cidr for %s : %s" %

```

Comment:
It used to read `log.error(blah %s and %s", (x, y))`. That fails because there are too few arguments. It needed to change to one of `log.error(blah %s and %s", x, y)` or `log.error(blah %s and %s" % (x, y))`. I chose the latter for no good reason whatsoever; I should be consistent, but it's really too late now. PEP8 probably says something about this, come to think of it. Next time I'll check